### PR TITLE
Docker broken links

### PR DESCRIPTION
**Pull request by catcecilia**
updated docker to remove broken links and replace them with updated links or wayback (L88, L91)

## What does this PR do?
Add resource(s) | Improve repo

### Description
updated docker to remove broken links and replace them with updated links or wayback (L88, L91)

### Why is this valuable (or not)?
valuable as these two links were 404 or missing

### How do we know it's really free?
easily accessible without a cost

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by catcecilia** [comment]
No longer has linter error

**Comment by eshellman** [comment]
the links you label as "markdown" are HTML, aren't they?

**Comment by catcecilia** [comment]
I believe I made a mistake and will be updating one to be HTML. the others I believe are markdown, but I could be wrong.

**Comment by eshellman** [comment]
you're wrong

**Comment by catcecilia** [comment]
I went ahead and fixed it.

**Comment by eshellman** [comment]
Sorry to have missed it before, but we're using 'html' in stead of 'online'

**Comment by catcecilia** [comment]
That's fine! I was trying to go off of what was already done prior but I went ahead and went with HTML instead of online

File: more/free-programming-cheatsheets.md [hunk 1]
```diff
@@ -85,18 +85,18 @@
...
 * [Docker Cheat Sheet](https://intellipaat.com/blog/tutorial/devops-tutorial/docker-cheat-sheet/) - IntelliPaat (HTML, PDF)
-* [Docker Cheat Sheet](https://swissarmydevops.com/containers/docker/docker-cli-cheat-sheet) - Nikko Pedersen, Swiss Army DevOps (HTML, PDF)
-* [Docker Cheat Sheet](https://github.com/wsargent/docker-cheat-sheet) - Will Sargent, et al.
+* [Docker Cheat Sheet](https://swissarmydevops.com/containers/docker/docker-cli-cheat-sheet) - Nikko Pedersen, Swiss Army DevOps (HTML, [PDF](https://web.archive.org/web/20210516172426/https://swissarmydevops.com/wp-content/uploads/2021/02/Docker_Cheat_Sheet-2.pdf)) *(:card_file_box: archived)*
+* [Docker Cheat Sheet](https://github.com/wsargent/docker-cheat-sheet) - Will Sargent, et al. (HTML)
 * [Docker Cheat Sheet - Kapeli](https://kapeli.com/cheat_sheets/Docker.docset/Contents/Resources/Documents/index) - Bogdan Popescu (HTML)
-* [Docker Cheat Sheet (:es:)](https://helmcode.com/posts/docker-cheat-sheet) - Cristian Córdova, HelmCode (HTML)
+* [Docker Cheat Sheet (:es:)](https://helmcode.com/blog/docker-cheat-sheet) - Cristian Córdova, HelmCode (HTML)
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:91]:
```suggestion
* [Docker Cheat Sheet (:es:)](https://helmcode.com/blog/docker-cheat-sheet) - Cristian Córdova, HelmCode (HTML)
```

File: more/free-programming-cheatsheets.md [hunk 2]
```diff
@@ -221,7 +221,7 @@
 
 * [MongoDB Cheat Sheet](https://www.mongodb.com/developer/quickstart/cheat-sheet) - MongoDB (HTML)
 * [MongoDB Cheat Sheet](https://blog.codecentric.de/files/2012/12/MongoDB-CheatSheet-v1_0.pdf) - codecentric (PDF)
-* [Quick Cheat Sheet for Mongo DB Shell commands](https://gist.github.com/michaeltreat/d3bdc989b54cff969df86484e091fd0c) - Michael Treat's Quick Cheat Sheet
+* [Quick Cheat Sheet for Mongo DB Shell commands](https://gist.github.com/michaeltreat/d3bdc989b54cff969df86484e091fd0c) - Michael Treat's Quick Cheat Sheet (Github Gist)
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:224]: Fix typos

```suggestion
* [Quick Cheat Sheet for Mongo DB Shell commands](https://gist.github.com/michaeltreat/d3bdc989b54cff969df86484e091fd0c) - Michael Treat's Quick Cheat Sheet (GitHub Gist)
```

**Review by davorpa — APPROVED** [review]

**Pull request merged**